packages/rum-core/src/domain/configuration/configuration.ts

```diff
@@ -296,6 +305,7 @@ export function serializeRumConfiguration(configuration: RumInitConfiguration) {
       track_user_interactions: configuration.trackUserInteractions,
       track_resources: configuration.trackResources,
       track_long_task: configuration.trackLongTasks,
+      plugins: configuration.plugins?.map((plugin) => assign({ name: plugin.name }, plugin.serializeConfiguration?.())),
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ What about doing the serialization within rum instead of the plugin? This way we can control how the serialization is done and might avoid duplicating serialization logic across multiple plugins

Reply:
Hum, 'serialization' might not be the best term for it. We don't serialize anything per-se, we transform the configuration before sending it as telemetry event. We need to ensure that we don't leak PII (ex: we avoid sending URLs specified in the configuration, ex: `configuration.proxy = 'https://...'` becomes `use_proxy: true`). It's not possible to define a generic way to do it.

Maybe we could rename `serializeConfiguration()` to something like `getConfigurationTelemetry()`?

Reply:
oh, I didn't understood it was for telemetry. `getConfigurationTelemetry` sounds good, even though it's a bit weird to expose this to public plugins, but I guess that's fine for now and we can revisit if needed when we go GA with plugins

Reply:
> it's a bit weird to expose this to public plugins

There is no plan to support public plugins any time soon though. We'll go GA with the react plugin, but this doesn't imply supporting third-party plugins.